test/yapi_network_test.rb

```diff
@@ -0,0 +1,374 @@
+        allow(Yast::Routing).to receive(:Write)
+      end
+
+      subject { Yast::YaPI::NETWORK.Write("route" => route) }
+
+      let(:success) { { "error" => "", "exit" => "0" } }
+
+      context "with no gateway in the default route" do
+        let(:route) { { "default" => nil } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with empty gateway in the default  route" do
+        let(:route) { { "default" => { "via" => "" } } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with a gateway in the default route" do
+        let(:route) { { "default" => { "via" => "TheIP" } } }
+        before do
+          expect(Yast::IP).to receive(:Check4).with("TheIP").and_return valid
+        end
+
+        context "if it's a valid IP4" do
+          let(:valid) { true }
+
+          it "returns success" do
+            expect(subject).to eq success
+          end
+
+          it "correctly modifies the routes" do
+            expect(Yast::Routing).to receive(:Routes=).with(
+              [
+                {
+                  "destination" => "default",
+                  "gateway"     => "TheIP",
+                  "netmask"     => "-",
+                  "device"      => "-"
+                }
+              ]
+            )
+            subject
+          end
+        end
+
+        context "if it's not a valid IP4" do
+          let(:valid) { false }
+
+          it "returns failure" do
+            res = subject
+            expect(res["exit"]).to eq "-1"
+            expect(res["error"]).to_not be_empty
+          end
+
+          it "doesnt't modify the routes" do
+            expect(Yast::Routing).to_not receive(:Routes=)
+            subject
+          end
+        end
+      end
+    end
+
+    context "with interfaces" do
+
+      subject { Yast::YaPI::NETWORK.Write("interface" => interface) }
+
+      before do
+        interface.keys.map do |k|
+          stub_clean_cache(k)
+        end
+      end
+
+      context("setting bootproto and startmode") do
+        let(:interface) { { "eth0" => { "bootproto" => "dhcp6", "startmode" => "onboot" } } }
+
+        it "sets both parameters correctly" do
+          expect(Yast::NetworkInterfaces).to receive(:Current=).with(
+            "BOOTPROTO" => "dhcp6",
+            "STARTMODE" => "onboot"
+          )
+          stub_write_interfaces
+          subject
+        end
+      end
+
+      context("setting IP address and netmask") do
+        let(:interface) { { "eth0" => { "ipaddr" => "TheIP/#{netmask}" } } }
+
+        context("with valid netmask in dot notation") do
```

Comment:
I just added what already was tested in old ones. At final it only parse the arguments and prepare it to <code>NetworkInterfaces->Current</code> so what i did was tested the correct parsing of them. 

It actually returns ever the same, so the output does not change even with wrong config. Just an example:
<code>
irb(main):028:0> Yast::YaPI::NETWORK.Write({"interface" => { "em1" => { "ipaddr" => "192.168.0.1" } } })
=> {"error"=>"", "exit"=>"0"}
irb(main):033:0> Yast.import 'SCR'
=> nil
irb(main):034:0> Yast::SCR.Read(Yast::Path.new('.network.value.em1.IPADDR'))
=> "192.168.0.1/192.168.0.1"
irb(main):035:0> Yast::YaPI::NETWORK.Write({"interface" => { "em1" => { "ipaddr" => "192.168.0.1.31" } } })
=> {"error"=>"", "exit"=>"0"}
irb(main):036:0> Yast::SCR.Read(Yast::Path.new('.network.value.em1.IPADDR'))
=> "192.168.0.1.31/192.168.0.1.31"
</code>

It is like the cookies monster that it whatever you give to him.